### ISSUE TITLE
Constantly failing dotnet watch tests on MacOS

### ISSUE THREAD

**Issue by jkotalik** [aspnet/AspNetCore#16912]
RestartProcessThatTerminatesAfterFileChange, RunsWithDotnetWatchEnvVariable, ChangeExcludedFile
```
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.AwaitableProcess.GetOutputLineAsync(String predicateName, Predicate`1 predicate, CancellationToken cancellationToken) in /_/src/Tools/dotnet-watch/test/AwaitableProcess.cs:line 100
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.AwaitableProcess.GetOutputLineAsync(String message, TimeSpan timeout) in /_/src/Tools/dotnet-watch/test/AwaitableProcess.cs:line 85
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.WatchableApp.StartWatcherAsync(String[] arguments, String name) in /_/src/Tools/dotnet-watch/test/Scenario/WatchableApp.cs:line 122
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.NoDepsAppTests.RestartProcessThatTerminatesAfterFileChange() in /_/src/Tools/dotnet-watch/test/NoDepsAppTests.cs:line 50
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by BrennanConroy** [comment]
Wasn't there an error about a missing SDK?

**Comment by jkotalik** [comment]
```
The temporary test folder is /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y
Copying project /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/TestProjects/NoDepsApp to /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp
Restoring msbuild project in NoDepsApp
Project directory: '/Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp'
Started process 8220: /Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet restore
  Restore completed in 96.6 ms for /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/NoDepsApp.csproj.


Process exited 8220
Process exited 8220 with code 0
Building NoDepsApp
Project directory: '/Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp'
Started process 8226: /Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet build
Microsoft (R) Build Engine version 16.4.0-preview-19460-01+e460c009a for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 26.52 ms for /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/NoDepsApp.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
  NoDepsApp -> /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/bin/Debug/netcoreapp5.0/NoDepsApp.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.79


Process exited 8226
Process exited 8226 with code 0
11/7/2019 9:20:17 PM: process start: '/Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tool/dotnet-watch.dll run --'
Waiting for output line [msg == 'Started']. Will wait for 120 sec.
11/7/2019 9:20:17 PM: post: 'watch : Running MSBuild target 'GenerateWatchList' on '/Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/NoDepsApp.csproj''
11/7/2019 9:20:17 PM: recv: 'watch : Running MSBuild target 'GenerateWatchList' on '/Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/NoDepsApp.csproj''. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:17 PM: post: 'watch : Started '/Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet' with process id 8233'
11/7/2019 9:20:17 PM: recv: 'watch : Started '/Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet' with process id 8233'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:19 PM: post: 'watch : Process id 8233 ran for 1930ms'
11/7/2019 9:20:19 PM: recv: 'watch : Process id 8233 ran for 1930ms'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:19 PM: post: 'watch : Watching 2 file(s) for changes'
11/7/2019 9:20:19 PM: recv: 'watch : Watching 2 file(s) for changes'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:19 PM: post: 'watch : Started '/Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet' with process id 8238'
11/7/2019 9:20:19 PM: recv: 'watch : Started '/Users/runner/runners/2.160.0/work/1/s/.dotnet/dotnet' with process id 8238'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:19 PM: post: 'watch : Running dotnet with the following arguments: run --'
11/7/2019 9:20:19 PM: recv: 'watch : Running dotnet with the following arguments: run --'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:19 PM: post: 'watch : Started'
11/7/2019 9:20:19 PM: recv: 'watch : Started'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:20 PM: post: 'Telemetry is: Disabled'
11/7/2019 9:20:20 PM: recv: 'Telemetry is: Disabled'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'Running /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/bin/Debug/netcoreapp5.0/NoDepsApp '
11/7/2019 9:20:24 PM: recv: 'Running /Users/runner/runners/2.160.0/work/1/s/artifacts/bin/dotnet-watch.Tests/Release/netcoreapp5.0/tmp/fsrjxp2x.c5y/NoDepsApp/bin/Debug/netcoreapp5.0/NoDepsApp '. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'Process ID: 8248'
11/7/2019 9:20:24 PM: recv: 'Process ID: 8248'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'It was not possible to find any compatible framework version'
11/7/2019 9:20:24 PM: recv: 'It was not possible to find any compatible framework version'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'The specified framework 'Microsoft.NETCore.App', version '5.0.0-alpha.1.19551.1' was not found.'
11/7/2019 9:20:24 PM: post: '  - Check application dependencies and target a framework version installed at:'
11/7/2019 9:20:24 PM: post: '      /Users/runner/.dotnet'
11/7/2019 9:20:24 PM: post: '  - Installing .NET Core prerequisites might help resolve this problem:'
11/7/2019 9:20:24 PM: recv: 'The specified framework 'Microsoft.NETCore.App', version '5.0.0-alpha.1.19551.1' was not found.'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '  - Check application dependencies and target a framework version installed at:'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      /Users/runner/.dotnet'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '  - Installing .NET Core prerequisites might help resolve this problem:'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: '      https://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409'
11/7/2019 9:20:24 PM: post: '  - The .NET Core framework and SDK can be installed from:'
11/7/2019 9:20:24 PM: post: '      https://aka.ms/dotnet-download'
11/7/2019 9:20:24 PM: post: '  - The following versions are installed:'
11/7/2019 9:20:24 PM: post: '      1.0.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.13 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.14 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.0.15 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      1.1.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.0.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.2.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.2.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.2.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: post: '      2.2.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'
11/7/2019 9:20:24 PM: recv: '      https://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '  - The .NET Core framework and SDK can be installed from:'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      https://aka.ms/dotnet-download'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '  - The following versions are installed:'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.13 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.14 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.0.15 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.10 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.11 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      1.1.12 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.0.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.4 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.5 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.6 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.7 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.8 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.1.9 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.2.0 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.2.1 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.2.2 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: recv: '      2.2.3 at [/Users/runner/.dotnet/shared/Microsoft.NETCore.App]'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'watch : Process id 8238 ran for 4773ms'
11/7/2019 9:20:24 PM: recv: 'watch : Process id 8238 ran for 4773ms'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'watch : Exited with error code 150'
11/7/2019 9:20:24 PM: recv: 'watch : Exited with error code 150'. Does not match condition '[msg == 'Started']'.
11/7/2019 9:20:24 PM: post: 'watch : Waiting for a file to change before restarting dotnet...'
11/7/2019 9:20:24 PM: recv: 'watch : Waiting for a file to change before restarting dotnet...'. Does not match condition '[msg == 'Started']'.
Disposing WatchableApp

```

**Comment by BrennanConroy** [comment]
There were no code/dependency changes between these tests passing, and them always failing. We should take a look and see if AzDo changed something?

https://docs.microsoft.com/en-us/azure/devops/release-notes/2019/sprint-160-update

**Comment by JunTaoLuo** [comment]
FYI @sebastienros

**Comment by anurse** [comment]
Does this repro on a local mac at all? @BrennanConroy can you try?

It's possible something changed in the SDK, or with how the host is resolving things.

**Comment by BrennanConroy** [comment]
> It's possible something changed in the SDK, or with how the host is resolving things.

Doubtful, we didn't get new bits or anything and it started failing. But I'll try it out.

**Comment by sebastienros** [comment]
It just passed: https://dev.azure.com/dnceng/public/_build/results?buildId=419353

**Comment by sebastienros** [comment]
Well, looks like it's marked a flaky test now

**Comment by anurse** [comment]
Let's see if we can repro it though. I'm fine with quarantine it it in The Flaky Zone to unblock things but I'd like to get to the bottom of this one.

**Comment by wtgodbe** [comment]
These tests are still failing in 3.1 & will block internal builds, should we mark them flaky/disable them there for now? How do we do that?

**Comment by wtgodbe** [comment]
See https://github.com/aspnet/AspNetCore/pull/16911

**Comment by anurse** [comment]
We can certainly mark them flaky in 3.1 as well. It's just a matter of putting the `Flaky` attribute on the test.

The fact that it's happening in 3.1 makes me more nervous. @BrennanConroy may be right that the machine environment changed. I wonder if these tests are running on Helix and if they're stable there?

**Comment by wtgodbe** [comment]
Maybe this is the same as https://github.com/dotnet/core-setup/pull/8802? Core-Setup was hitting some similar failures on OSX yesterday.

**Comment by wtgodbe** [comment]
These failures look _very_ similar to https://github.com/dotnet/core-setup/pull/8775#issuecomment-551306746, so probably we can do something similar to https://github.com/dotnet/core-setup/pull/8802 to fix it.

**Comment by wtgodbe** [comment]
I put up https://github.com/aspnet/AspNetCore/pull/16932 do disable these in 3.1 in the meantime (Looks like they're already skipped in master)

https://github.com/aspnet/AspNetCore/blob/845e6d5512c3521a07d7fa93737b0da041a8a3c6/src/Tools/dotnet-watch/test/NoDepsAppTests.cs#L45

**Comment by anurse** [comment]
> These failures look _very_ similar to [dotnet/core-setup#8775 (comment)](https://github.com/dotnet/core-setup/pull/8775#issuecomment-551306746), so probably we can do something similar to [dotnet/core-setup#8802](https://github.com/dotnet/core-setup/pull/8802) to fix it.

Nice find!

**Comment by wtgodbe** [comment]
> Nice find!

[My thoughts exactly 😆 ](https://github.com/dotnet/core-setup/pull/8775#issuecomment-551310415)

**Comment by wtgodbe** [comment]
https://github.com/aspnet/AspNetCore/pull/16949

**Comment by BrennanConroy** [comment]
Fixed in 3.0, 3.1, blazor-wasm, and master.

I think we can close
